hawkular-inventory-paths/src/main/java/org/hawkular/inventory/paths/Path.java

```diff
@@ -1208,17 +1212,17 @@ public Impl get() {
 
         @SuppressWarnings("unchecked")
         public This resource(String id) {
-            segments.add(new Segment(Resource.class, id));
+            segments.add(new Segment(SegmentType.r, id));
             return (This) this;
         }
 
-        public SDB data(Resources.DataRole role) {
-            segments.add(new Segment(DataEntity.class, role.name()));
+        public SDB data(String role) {
```

Comment:
> Do you have an idea how we could keep the previous nice behavior?

No, I was hoping you could come up with an idea :) 

I am still far from understanding what these roles are exactly for. 

At the first sight, it looked too complicated to take them to the new path module, because they pulled other classes with them. Within the new paths module, the role enums are given by the SegmentType. So maybe we (a) could create some minimal Role enums in the path module that do not pull anything with them? 

Alternativelly, (b) wrapping these role-related method calls somehow on the inventory side sounds both complicated and actually also useless, because every consumer of the new paths package should take the advantage of the type safety.